pkg/awx/connection.go

```diff
@@ -42,6 +43,10 @@ type ConnectionBuilder struct {
 	agent    string
 	token    string
 	insecure bool
+
+	// If cacrt is specified, the client will expect the server certificate
+	// to be signed by this CA. If it isn't, the client will use the host trust store.
```

Comment:
Note that the server certificate doesn't need to be directly signed by this CA, it just needs to be verifiable using these CA certificates (there can be multiple certificates) as the roots. Not really important, just wanted to make sure that you understand the difference.